Source/WebCore/bindings/js/SerializedScriptValue.cpp

```diff
@@ -2959,31 +2966,33 @@ class CloneDeserializer : CloneBase {
             readLittleEndian(ptr, end, c);
             characters[i] = c;
         }
+        if (shouldAtomize == ShouldAtomize::Yes)
+            AtomString { str };
```

Comment:
I think you want to write `str = AtomString { str }` here. But also, if this is at all common, we can write a more optimized version of this for short strings that doesn’t involve allocating and then deallocating memory when it’s a duplicate atom.

Reply:
As far as I know, both x86-64 and Apple Silicon use little-endian so I didn't spend time optimizing this code path, though we could fairly easily do so.